includes/widgets/icon.php

```diff
@@ -331,6 +352,27 @@ protected function register_controls() {
 			]
 		);
 
+		$this->add_control(
+			'icon_fit_to_size',
+			[
+				'label'       => esc_html__( 'Fit to Size', 'elementor' ),
+				'type'        => Controls_Manager::SLIDER,
+				'selectors'   => [
+					'{{WRAPPER}} .elementor-icon' => 'padding: {{SIZE}}{{UNIT}};',
+				],
+				'range'       => [
+					'em' => [
+						'min' => 0,
+						'max' => 5,
+					],
+				],
+				'condition'   => [
+					'view!' => 'default',
+				],
+				'render_type' => 'template',
```

Comment:
I am not convinced you need this, not in this control nor in the other one. Please try to remove the `render_type` properties from both controls and test whether it works or not.

Reply:
This is because any control which has a `'selectors'` property automatically is treated as a "CSS control".
"CSS controls" in general do not trigger a re-render on change, by design, since live CSS updates in the Editor are handled in real time in Javascript.